lib/features/leaderboard/domain/usecases/get_leaderboard.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:coffeecard/core/errors/failures.dart';
+import 'package:coffeecard/core/usecases/usecase.dart';
+import 'package:coffeecard/features/leaderboard/data/datasources/leaderboard_remote_data_source.dart';
+import 'package:coffeecard/features/leaderboard/domain/entities/leaderboard_user.dart';
+import 'package:coffeecard/features/leaderboard/presentation/cubit/leaderboard_cubit.dart';
+import 'package:dartz/dartz.dart';
+
+class GetLeaderboard
+    implements UseCase<List<LeaderboardUser>, LeaderboardFilter> {
+  final LeaderboardRemoteDataSource remoteDataSource;
+
+  GetLeaderboard({required this.remoteDataSource});
+
+  @override
+  Future<Either<Failure, List<LeaderboardUser>>> call(
+    LeaderboardFilter filter,
+  ) async {
+    final userleaderboard = await remoteDataSource.getLeaderboardUser(filter);
```

Comment:
Better naming could benefit this code
```suggestion
    final user = await remoteDataSource.getLeaderboardUser(filter);
```
```suggestion
    final leaderboardUser = await remoteDataSource.getLeaderboardUser(filter);
```